contracts/src/v0.8/dev/ocr2dr/OCR2DR.sol

```diff
@@ -0,0 +1,249 @@
+  }
+
+  enum HttpVerb {
+    Get
+    // In future version we may add other verbs
+  }
+
+  struct HttpHeader {
+    string key;
+    string value;
+  }
+
+  struct HttpQuery {
+    HttpVerb verb;
+    string url;
+    HttpHeader[] headers;
+  }
+
+  struct Request {
+    Location codeLocation;
+    Location secretsLocation;
+    CodeLanguage language;
+    string source; // Source code for Location.Inline or url for Location.Remote
+    bytes secrets; // Encrypted secrets blob for Location.Inline or url for Location.Remote
+    string[] args;
+    HttpQuery[] queries;
+  }
+
+  /**
+   * @notice Encodes a Request to CBOR encoded bytes
+   * @param self The request to encode
+   * @return CBOR encoded bytes
+   */
+  function encodeCBOR(Request memory self) internal pure returns (bytes memory) {
+    BufferChainlink.buffer memory buf;
+    BufferChainlink.init(buf, defaultBufferSize);
+
+    buf.encodeString("codeLocation");
+    buf.encodeUInt(uint256(self.codeLocation));
+
+    buf.encodeString("language");
+    buf.encodeUInt(uint256(self.language));
+
+    buf.encodeString("source");
+    buf.encodeString(self.source);
+
+    if (self.args.length > 0) {
+      buf.encodeString("args");
+      buf.startArray();
+      for (uint256 i = 0; i < self.args.length; i++) {
+        buf.encodeString(self.args[i]);
+      }
+      buf.endSequence();
+    }
+
+    if (self.secrets.length > 0) {
+      buf.encodeString("secretsLocation");
+      buf.encodeUInt(uint256(self.secretsLocation));
+      buf.encodeString("secrets");
+      buf.encodeBytes(self.secrets);
+    }
+
+    if (self.queries.length > 0) {
+      buf.encodeString("queries");
+      buf.startArray();
+      for (uint256 i = 0; i < self.queries.length; i++) {
+        buf.startMap();
+        buf.encodeString("verb");
+        buf.encodeUInt(uint256(self.queries[i].verb));
+        buf.encodeString("url");
+        buf.encodeString(self.queries[i].url);
+        if (self.queries[i].headers.length > 0) {
+          buf.encodeString("headers");
+          buf.startMap();
+          for (uint256 j = 0; j < self.queries[i].headers.length; j++) {
+            buf.encodeString(self.queries[i].headers[j].key);
+            buf.encodeString(self.queries[i].headers[j].value);
+          }
+          buf.endSequence();
+        }
+        buf.endSequence();
+      }
+      buf.endSequence();
+    }
+
+    return buf.buf;
+  }
+
+  /**
+   * @notice Initializes a OCR2DR Request
+   * @dev Sets the codeLocation and code on the request
+   * @param self The uninitialized request
+   * @param location The user provided source code location
+   * @param language The programming language of the user code
+   * @param source The user provided source code or a url
+   * @return The initialized request
+   */
+  function initializeRequest(
+    Request memory self,
+    Location location,
+    CodeLanguage language,
+    string memory source
+  ) internal pure returns (OCR2DR.Request memory) {
```

Comment:
not sure if you really need the return parameter here since you are changing self itself. I think if you declare a return type a new object is allocated to be returned

Reply:
The intention was to provide chaining, e.g. `initializeRequest(...).addQuery(...)`, but I do agree this does not make it elegant. Removed.